src/main/java/com/github/davidmoten/rtree/RTree.java

```diff
@@ -879,51 +782,39 @@ public Visualizer visualize(int width, int height, Rectangle view) {
      * restricted to the the smallest view that fully contains the coordinates. The
      * points in the view are scaled to match the aspect ratio defined by the width
      * and height.
-     * 
-     * @param width
-     *            of the image in pixels
-     * @param height
-     *            of the image in pixels
+     *
+     * @param width  of the image in pixels
+     * @param height of the image in pixels
      * @return visualizer
      */
     public Visualizer visualize(int width, int height) {
         return visualize(width, height, calculateMaxView(this));
     }
 
     private Rectangle calculateMaxView(RTree<T, S> tree) {
-        return tree.entries().reduce(Optional.<Rectangle>absent(),
-                new Func2<Optional<Rectangle>, Entry<T, S>, Optional<Rectangle>>() {
-
-                    @Override
-                    public Optional<Rectangle> call(Optional<Rectangle> r, Entry<T, S> entry) {
-                        if (r.isPresent())
-                            return of(r.get().add(entry.geometry().mbr()));
-                        else
-                            return of(entry.geometry().mbr());
-                    }
-                }).toBlocking().single().or(rectangle(0, 0, 0, 0));
+        return tree.entries()
+                .reduce(Optional.empty(), rectangleAccumulator)
+                .toBlocking().single()
+                .orElse(ZERO_RECTANGLE);
     }
 
     public Optional<? extends Node<T, S>> root() {
         return root;
     }
 
     /**
-     * If the RTree has no entries returns {@link Optional#absent} otherwise returns
+     * If the RTree has no entries returns {@link Optional#empty()} otherwise returns
      * the minimum bounding rectangle of all entries in the RTree.
-     * 
+     *
      * @return minimum bounding rectangle of all entries in RTree
      */
     public Optional<Rectangle> mbr() {
-        if (!root.isPresent())
-            return absent();
-        else
-            return of(root.get().geometry().mbr());
+        return root.map(r -> r.geometry().mbr());
```

Comment:
This increases allocation costs for this call, please don't do it.

Reply:
It's not clear for me. A new Optional object is to be created independently on the condition, and the lambda will be executed only if the root is not empty, won't it?

Reply:
Every time this method is an object representing the lambda is created. This is what I mean by an allocation cost.

Reply:
I did a bit of research and found this statement in https://stackoverflow.com/questions/27524445/does-a-lambda-expression-create-an-object-on-the-heap-every-time-its-executed:

>The behavior is not exactly specified. The JVM is given big freedom on how to implement it. Currently, Oracle’s JVM creates (at least) one instance per lambda expression (i.e. doesn’t share instance between different identical expressions) but creates singletons for all expressions which don’t capture values.

I'm happy with the lambdas then, thanks.